docs/book/src/topics/ipv6-enabled-cluster.md

```diff
@@ -0,0 +1,367 @@
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - fd01::/48
+    services:
+      cidrBlocks:
+      - fd02::/112
+```
+
+<aside class="note warning">
+
+<h1>Warning</h1>
+
+**DNS64/NAT64**: If you are configuring CAPA to create dualstack private subnets (by default) for an IPv6 cluster and need IPv6-only pods to reach IPv4-only internet services, you must enable [DNS64/NAT64](https://docs.aws.amazon.com/vpc/latest/userguide/nat-gateway-nat64-dns64.html) as CAPA does not do so.
+
+**CoreDNS**: Since CoreDNS pods run on the single-stack IPv6 pod network, they will fail to resolve non-cluster DNS queries via the IPv4 upstream nameserver in `/etc/resolv.conf`.
+
+The workaround is to edit the `coredns` ConfigMap in namespace `kube-system` to use Route53 Resolver IPv6 nameserver `fd00:ec2::253`, by setting `forward . /etc/resolv.conf` part to `forward . fd00:ec2::253 /etc/resolv.conf`.
+  ```bash
+  kubectl -n kube-system edit cm/coredns
+  ```
+
+**Note**: This CoreDNS workaround is NOT required for dualstack clusters where pods have both IPv4 and IPv6 addresses.
+
+</aside>
+
+## Creating Dualstack Self-managed Clusters
+
+To quickly deploy a dualstack self-managed cluster, use the [Dualstack cluster template](https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-aws/refs/heads/main/templates/cluster-template-dualstack.yaml).
+
+When creating a self-managed cluster, you can define both IPv4 and IPv6 Pod and Service CIDRs. For example:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - 192.168.0.0/16
+      - fd01::/48
+    services:
+      cidrBlocks:
+      - 172.30.0.0/16
+      - fd02::/112
+```
+
+## Cloud Controller Manager IPv6 Support for Self-managed Clusters
+
+<aside class="note warning">
+<h1>Warning</h1>
+
+The AWS Cloud Controller Manager (CCM) does not currently support dualstack Load Balancers. When creating Services of type LoadBalancer in a dualstack cluster, they will be
+assigned addresses from only one IP family.
+</aside>
+
+**Node IP addresses**: You need to provide cloud-config to the CCM via a ConfigMap to set the `NodeIPFamilies` to include IPv6. This instructs the CCM to consider IPv6 in the node's network interface. If not, the CCM will only consider node's IPv4. This causes nodes to have only IPv4 and new pods with `hostNetwork: true` will only pick up the node's IPv4 address.
+
+For example, provide the following ConfigMap to `cloud-controller-manager-addon`:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cloud-config
+  namespace: kube-system
+data:
+  cloud-config.conf: |
+    [Global]
+    NodeIPFamilies=ipv4
+    NodeIPFamilies=ipv6
```

Comment:
This is a little confusing - L210-211 says that only one IP family can be used on a load balancer, but this example shows a cloud.conf with both.

We might want to reword L210 to explicitly say that this configures the CCM's to create dedicated LBs for each family.

Reply:
Thanks, addressed! I separated the mentioning of node IP and Service LB (managed by CCM) into separate sections now.